### PR TITLE
[Common] TrackQAConverter002- for TOF modification

### PR DESCRIPTION
**Pull request by miranov25**
# Relates to the O2 modification - https://github.com/AliceO2Group/AliceO2/pull/13787

### CODE REVIEW

**Comment by miranov25** [comment]
Code modification provided by @f3sch.  It will need O2 modifications merged before.

**Comment by vkucera** [comment]
@pzhristov The PR does not compile. Please revert.

**Comment by miranov25** [comment]
Hello @vkucera and @pzhristov,

This pull request requires the O2 repository to be updated. The O2 update was merged a few minutes ago.

In my case, everything works well once both repositories are synchronized. What is the usual strategy for syncing O2 and O2Physics? @f3sch mentioned that we might need to wait for some time.

Best regards,
Marian

**Comment by ktf** [comment]
In general we first merge in O2, do a build and then merge the O2Physics changes. @singiamtel can you trigger a new nightly to fix this? Thanks.

File: Common/TableProducer/Converters/trackQA002Converter.cxx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+using namespace o2;
+using namespace o2::framework;
+
+struct TrackQAConverter002 {
+  Produces<aod::TracksQA_002> tracksQA_002;
+
+  void process000(aod::TracksQA_000 const& tracksQA_002)
+  {
+    for (const auto& trackQA : tracksQA_000) {
+      tracksQA_002(
```
Comment (vkucera) [inline Common/TableProducer/Converters/trackQA002Converter.cxx:26]: Sorry, but this is clearly a bug which does not compile.

File: Common/TableProducer/Converters/trackQA002Converter.cxx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        std::numeric_limits<int8_t>::min(),  // dTofdX
+        std::numeric_limits<int8_t>::min()); // dTofdY
+    }
+  }
+  PROCESS_SWITCH(TrackQAConverter002, process000, "process v000-to-v002 conversion", false);
+
+  void process001(aod::TracksQA_001 const& tracksQA_002)
+  {
+    for (const auto& trackQA : tracksQA_001) {
+      tracksQA_002(
```
Comment (vkucera) [inline Common/TableProducer/Converters/trackQA002Converter.cxx:60]: Same here

File: Common/TableProducer/Converters/trackQA002Converter.cxx (hunk 1 above)
Comment (miranov25) [reply Common/TableProducer/Converters/trackQA002Converter.cxx:26]: Thank you for checking.  I got the modification from Felix, and I assume that I compiled it.
Looks like it was not synced to my build server properly

**Comment by miranov25** [comment]
Hello @vkucera 
I see.  I commented above.
>Thank you for checking. I got the modification from Felix, and I assume that I compiled it.
Looks like it was not synced to my build server properly. 

Please wait for change

**Comment by miranov25** [comment]
Hello @vkucera,

The code has compiled successfully with the fix applied above. It will take another 15 minutes to complete the full compilation (converter was already compiled).

Since I haven’t tested the converter before, could you advise on the recommended way to test it? 

I typically work with AO2D files in ROOT sessions and Python, using derived data sets,  but I’m not sure how to properly validate the converter.

Best regards,
Marian

**Comment by vkucera** [comment]
Hi @miranov25 , thanks, I confirm that O2Physics compiles with the fix above. Please make a PR.

**Comment by miranov25** [comment]
Can I reuse this pull request, or should I create new one?

**Comment by vkucera** [comment]
> Can I reuse this pull request, or should I create new one?

This PR has been merged. You have to make a new one, although you can use the same branch.

**Comment by miranov25** [comment]
Creating pull request with fix - https://github.com/AliceO2Group/O2Physics/pull/8961

**Comment by vkucera** [comment]
> Creating pull request with fix - #8961

Thanks

**Comment by ddobrigk** [comment]
It would sense to click revert and force-merge the revert - could we do that, @pzhristov ? We anyway cannot merge the fix right now...

**Pull request merged**